packages/kit/src/runtime/client/client.js

```diff
@@ -753,8 +752,10 @@ function has_changed(
 		if (params[param] !== current.params[param]) return true;
 	}
 
-	for (const href of uses.dependencies) {
-		if (invalidated.some((fn) => fn(new URL(href)))) return true;
+	if (!parent_changed) {
```

Comment:
Why would this logic only run if the parent hasn't changed? Wouldn't this result in false negatives?

Reply:
What I understand of the whole thing is that every "parent routes" needs to go through the whole invalidation loop else their version of data is not updated. The logic in the file starts with the current page, then up a level until there is none. 

The fix in https://github.com/sveltejs/kit/pull/11268 would strictly break this loop after first iteration, so parents are never actually loaded. 

Having `!parent_changed` there is a mean  to ensure this is not the first iteration of the loop (meaning it's a parent), so it's data isn't actually invalidated (no need) but every routes still go through a invalidation process (so they get the updated data). 

So basically there is a `invalidation process` that do several things (invalidate the data and assign this data to the route being invalidated), and every segment of the page **need** to go through this process to receive the proper data, but only the current page segment have to actually invalidate data.

If you prevent parent segments from going into this loop they never get the updated data.